From 9af7b3dfac265827eac8b7c9b9017a74b64f97d2 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 8 Jun 2017 15:53:44 +0000
Subject: [PATCH 0847/1077] MANTIS 0045407 Transmit Billing Data records to a
 configured WEB Service. Mantis 0046042: CR 030 PNR Reporting

PlatformIF
added logCommand for logging more printing details

removed
logATBResponse
logBTPResponse
logGPPResponse
logATBNonAEAResponse
logBTPNonAEAResponse
logGPPNonAEAResponse
logCONResponse

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1208 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/platform.idl            | 69 ++++++++-----------
 1 file changed, 28 insertions(+), 41 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/platform.idl b/SelfServiceCommon/Massai/idl/Massai/platform.idl
index b83faa75..0c65cf4a 100644
--- a/SelfServiceCommon/Massai/idl/Massai/platform.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/platform.idl
@@ -228,8 +228,33 @@ module PlatformManagement
       long mediaCount;
       string type;               // type = "gpp" or type ="atb"; needed for evaluation of mediaStatusTypeValue
     };
+
+    enum DeviceType
+    {
+      GPP,
+      ATB,
+      BPP,
+      BTP
+    };
     
-    
+    enum DataType
+    {
+      AEA,
+      SVG,
+      PDF
+    };
+
+    struct LoggingCommandData
+    {
+      string unitId;
+      long couponsproduced;
+      DeviceType type;          
+      DataType dataType; 
+      string request;
+      string response;
+      unsigned long long timestamp;        // utc time; number of milli seconds since 1.1.1970 00:00
+    };
+        
     enum ApplicationsState
     {
       APP_INITIALIZE,                 // application is initializing or loading
@@ -256,11 +281,11 @@ module PlatformManagement
     
     interface PlatformIF
     {
-
         typedef sequence<LoggingComponentData> LoggingComponentDataList;   
         
         void initLogging(in string realComponentName, in LoggingComponentDataList init);
         void logCounter(in LoggingCounterData lcd);
+        void logCommand(in LoggingCommandData command);
         void logCounterFailure(in string unitId, in LoggingClass unitClass, in long numberOfFailed);
         void updateMediaCount(in string compRef, in long mediaCount);
         void logCounterSuccess(in string unitId, in LoggingClass unitClass, in long numberOfSuccesfullDocs);
@@ -282,45 +307,7 @@ module PlatformManagement
         /** Gets the currently active token.
             @return the currently active token. */
         string getActiveToken();
-
-        /** Logs an ATB response for billing purposes.
-            @param response - The ATB response
-            @return RC_OK. */
-        code logATBResponse(in string response);
-
-        /** Logs an BTP response for billing purposes.
-            @param response - The BTP response
-            @return RC_OK. */
-        code logBTPResponse(in string response);
-
-        /** Logs an GPP response for billing purposes.
-            @param response - The GPP response
-            @return RC_OK.*/
-        code logGPPResponse(in string response);
-
-        /** Logs an ATB response for billing purposes.
-            @param type - the type of the print command
-            @param used - the number of documents printed
-            @return RC_OK. */
-        code logATBNonAEAResponse(in string type, in long used);
-
-        /** Logs an GPP response for billing purposes.
-            @param type - the type of the print command
-            @param used - the number of documents printed
-            @return RC_OK. */
-        code logGPPNonAEAResponse(in string type, in long used);
-
-        /** Logs an BTP response for billing purposes.
-            @param type - the type of the print command
-            @param used - the number of documents printed
-            @return RC_OK. */
-        code logBTPNonAEAResponse(in string type, in long used);
-
-        /** Logs an conveyor response for billing purposes.
-            @param delivered - The number of baggages delivered
-            @return RC_OK. */
-        code logCONResponse(in long delivered);
-
+        
         /** Dispatches a platform event.
             @param event - The plaform event
             @return RC_OK if no error occurs, RC_ERROR if the event has the wrong format. */
-- 
2.41.0.windows.1

